File: go/challenges/anagrams_test.go
package challenges

import (
	"testing"
)

func TestCheckAnagrams(t *testing.T) {
	t.Run("expect to get an error if strings have different length", func(t *testing.T) {
		firstString := "awesome"
		secondString := "awesom"

		err := CheckAnagrams(firstString, secondString)

		assertError(t, err, ErrDifferentLengths)
	})

	t.Run("expect to get an error if the strings have different letter", func(t *testing.T) {
		firstString := "car"
		secondString := "rat"

		err := CheckAnagrams(firstString, secondString)

		assertError(t, err, ErrStringHasDiffLetter)
	})

	t.Run("expect to get an error if the strings have different letter frequency", func(t *testing.T) {
		firstString := "aaz"
		secondString := "zza"

		err := CheckAnagrams(firstString, secondString)

		assertError(t, err, ErrStringHasDiffLetterFreq)
	})

	t.Run("should not return when input is correct anagram", func(t *testing.T) {
		firstString := "anagram"
		secondString := "nagaram"

		err := CheckAnagrams(firstString, secondString)

		if err != nil {
			t.Errorf("expected to get no error, but got %q", err)
		}
	})
}

func assertError(t *testing.T, got, want error) {
	t.Helper()

	if got == nil {
		if want == nil {
			return
		}
		t.Fatal("expected to get an error")
	}

	if got != want {
		t.Errorf("got %q, want %q", got, want)
	}
}

File: javascript/src/challenges/anagrams.test.ts
import { validAnagram } from './anagrams'

describe('Anagrams Check Challenge', () => {
    it('should return true if empty string', () => {
        const firstString = ''
        const secondString = ''

        const isValid = validAnagram(firstString, secondString)

        expect(isValid).toBeTruthy()
    })

    it('should return false if the length is different', () => {
        const firstString = 'awesome'
        const secondString = 'awesom'

        const isValid = validAnagram(firstString, secondString)

        expect(isValid).toBeFalsy()
    })

    it('should return true if pass correct anagram', () => {
        const firstString = 'anagram'
        const secondString = 'nagaram'

        const isValid = validAnagram(firstString, secondString)

        expect(isValid).toBeTruthy()
    })

    it('should return false if the frequency of letters are not same', () => {
        const firstString = 'aaz'
        const secondString = 'zza'

        const isValid = validAnagram(firstString, secondString)

        expect(isValid).toBeFalsy()
    })

    it('should return false for "rat" and "car"', () => {
        const firstString = 'rat'
        const secondString = 'car'

        const isValid = validAnagram(firstString, secondString)

        expect(isValid).toBeFalsy()
    })

    it('should return true for "qwerty" and "qeywrt"', () => {
        const firstString = 'qwerty'
        const secondString = 'qeywrt'

        const isValid = validAnagram(firstString, secondString)

        expect(isValid).toBeTruthy()
    })

    it('should return true for "texttwisttime" and "timetwisttext"', () => {
        const firstString = 'texttwisttime'
        const secondString = 'timetwisttext'

        const isValid = validAnagram(firstString, secondString)

        expect(isValid).toBeTruthy()
    })
})
